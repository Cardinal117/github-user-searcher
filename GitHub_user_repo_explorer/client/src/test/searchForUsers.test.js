import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import SearchForUsers from "../components/searchForUsers";

// Test for SearchForUsers component to see if it is being rendered correctly.
describe("SearchForUsers Component", () => {
  it("renders input and button correctly", () => {
    render(
      <MemoryRouter>
        <SearchForUsers />
      </MemoryRouter>
    );
    // Check if the input and button is in the document.
    expect(screen.getByPlaceholderText(/enter username/i)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /search/i })).toBeInTheDocument();
  });

  it("accepts user input: Cardinal117", () => {
    render(
      <MemoryRouter>
        <SearchForUsers />
      </MemoryRouter>
    );
    // Find the input field according to placeholder text and simulate the user adding a value.
    const input = screen.getByPlaceholderText(/enter username/i);
    fireEvent.change(input, { target: { value: "Cardinal117" } });
    expect(input.value).toBe("Cardinal117");
  });

  // Test for loading state to see if it is rendered correctly.
  it("renders loading state correctly", () => {
    render(
      <MemoryRouter>
        <SearchForUsers externalTestLoading={true} />
      </MemoryRouter>
    );
    // Check if loading message appears.
    expect(screen.getByText(/loading/i)).toBeInTheDocument();
  });

  // Test for error state to see if it is rendered correctly.
  it("renders error state correctly", () => {
    render(
      <MemoryRouter>
        <SearchForUsers externalTestError={true} />
      </MemoryRouter>
    );
    // Check if error message appears.
    expect(screen.getByText(/error/i)).toBeInTheDocument();
  });
});
