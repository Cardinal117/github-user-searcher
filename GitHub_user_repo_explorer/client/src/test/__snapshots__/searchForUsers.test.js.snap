// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchForUsers component correctly 1`] = `null`;

exports[`renders SearchForUsers component error state correctly 1`] = `null`;

exports[`renders SearchForUsers component loading state correctly 1`] = `null`;
